src/types.ts

```diff
@@ -0,0 +1,52 @@
+export type NewRelicOptions = {
+  tracing?: NewrelicTracingApi,
+  segment?: NewrelicSegmentApi,
+  metrics?: NewrelicMetricsApi,
+  debug: boolean,
+};
+
+export type Spec = Record<any, any>
+export type Nullable<T> = T | null
+
+export interface NewrelicTracingApi {
+  enabled: boolean,
+  accountApiKey: string,
+  serviceName: string,
+}
+
+export interface NewrelicSegmentApi {
+  enabled: boolean,
+}
+
+export interface NewrelicMetricsApi {
+  enabled: boolean,
+  accountApiKey: string,
+}
+
+export interface TelemetryCollector {
+  API_KEY: string,
```

Comment:
if these are configuration options, they should be definable as plugin options under
`const defaults: NewRelicOptions = {` (which is a Gubu spec too :) )